ARK Server Manager/Lib/ModUpdater/ModCopy.cs

```diff
@@ -0,0 +1,303 @@
+            if (string.IsNullOrWhiteSpace(steamCmdPath) || string.IsNullOrWhiteSpace(serverPath) || ids == null || ids.Length == 0)
+                return false;
+
+            if (!Directory.Exists(steamCmdPath))
+                return false;
+
+            try
+            {
+                foreach (string id in ids)
+                {
+                    string workshopModPath = Path.Combine(steamCmdPath, @"steamapps\workshop\content\346110\", id);
+
+                    if (Directory.Exists(workshopModPath))
+                    {
+                        string fileName1 = Path.Combine(workshopModPath, "mod.info");
+                        List <string> list = new List<string>();
+                        ParseBaseInformation(fileName1, list);
+
+                        string fileName2 = Path.Combine(workshopModPath, "modmeta.info");
+                        Dictionary<string, string> metaInformation = new Dictionary<string, string>();
+                        if (ParseMetaInformation(fileName2, metaInformation))
+                            workshopModPath = Path.Combine(workshopModPath, "WindowsNoEditor");
+
+                        string serverModPath = Path.Combine(serverPath, @"ShooterGame\Content\Mods\", id);
+                        bool flag = false;
+
+                        foreach (string workshopModFile in Directory.GetFiles(workshopModPath, "*.*", SearchOption.AllDirectories))
+                        {
+                            string serverModFile = workshopModFile.Replace(workshopModPath, serverModPath);
+                            string serverModFilePath = Path.GetDirectoryName(serverModFile);
+
+                            if (!Directory.Exists(serverModFilePath))
+                                Directory.CreateDirectory(serverModFilePath);
+
+                            string serverModFileExtension = Path.GetExtension(workshopModFile).ToUpper();
+                            if (Path.GetFileNameWithoutExtension(workshopModFile).Contains("PrimalGameData"))
+                                flag = true;
+
+                            if (string.Compare(serverModFileExtension, ".uncompressed_size", StringComparison.OrdinalIgnoreCase) != 0)
+                            {
+                                if (string.Compare(serverModFileExtension, ".z", StringComparison.OrdinalIgnoreCase) == 0)
+                                    UE4ChunkUnzip(workshopModFile, serverModFile.Substring(0, serverModFile.Length - 2));
+                                else
+                                    File.Copy(workshopModFile, serverModFile, true);
+                            }
+                        }
+
+                        if (metaInformation.Count == 0 && flag)
+                            metaInformation["ModType"] = "1";
+
+                        WriteModFile($"{serverModPath}.mod", id, metaInformation, list);
+                    }
+                }
+                return true;
+            }
+            catch
```

Comment:
Is it possible to get exceptions in the normal course, and should we log something or bubble up an error message?  Unconditional catches are usually an indication of missing failure case handling.